libs/@local/hashql/core/src/type/kind/generic/mod.rs

```diff
@@ -23,8 +36,8 @@ newtype!(
 
 newtype_producer!(pub struct GenericArgumentIdProducer(GenericArgumentId));
 
-// The name is stored in the environment, to allow for `!Drop`
-#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash)]
+// PartialEq, Eq, PartialOrd, Ord, Hash are derived, with `name` ignored
```

Comment:
I see why you do this, however, I'm not particularly a fan of this. To me, equality in 99% of the case means structural equality. Leaving out a field for comparison may result in really unexpected behavior. 
Do you see a chance in this being changed?
For equality, we could implement an `equiv(&self, &Self) -> bool` or something. For `Hash`, we might want to implement it on another type only consisting of `id` and `constraint` and use that as Hash key.

Reply:
I have had the same thought. Primarily considered these because of the `partition_dedupe` and `sort_unstable` calls when creating the types. Maybe it'd make sense to create a new struct, call it: `AnonymousGenericArgument` and have a function like `as_anonymous` to create it. Considering that we're only shuffling around ~12 bytes (4 bytes `GenericArgumentId`, 4 bytes `TypeId`, 4 bytes overhead because of `Option`) this should be fine.

What do you think?

Reply:
Yes, let's go for that

Reply:
done in [b48290b](https://github.com/hashintel/hash/pull/7162/commits/b48290b51764a50aef9bbf15b9544fd2212063e8)

Reply:
Thank you!